spec/json.go

```diff
@@ -0,0 +1,109 @@
+package spec
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+)
+
+type JSONSpec Spec
+
+func (spec Spec) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		JSONSpec
+		Version string `json:"openapi"`
```

Comment:
Nice trick, but you can define it inside the function. 
```suggestion
func (spec Spec) MarshalJSON() ([]byte, error) {
	type JSONSpec Spec
	return json.Marshal(struct {
		JSONSpec
		Version string `json:"openapi"`
```